content/ember/v3/ember-built-in-components-legacy-attribute-arguments.md

```diff
@@ -0,0 +1,337 @@
+  * `@key-down` (for the `keydown` event)
+  * `@key-press` (for the `keypress` event)
+  * `@key-up` (for the `keyup` event)
+* `<Textarea>`
+  * `@click`
+  * `@contextMenu` (for the `contextmenu` event)
+  * `@doubleClick` (for the `dblclick` event)
+  * `@drag`
+  * `@dragEnd` (for the `dragend` event)
+  * `@dragEnter` (for the `dragenter` event)
+  * `@dragLeave` (for the `dragleave` event)
+  * `@dragOver` (for the `dragover` event)
+  * `@dragStart` (for the `dragstart` event)
+  * `@drop`
+  * `@input`
+  * `@mouseDown` (for the `mousedown` event)
+  * `@mouseEnter` (deprecated, for the `mouseenter` event)
+  * `@mouseLeave` (deprecated, for the `mouseleave` event)
+  * `@mouseMove` (deprecated, for the `mousemove` event)
+  * `@mouseUp` (for the `mouseup` event)
+  * `@submit`
+  * `@touchCancel` (for the `touchcancel` event)
+  * `@touchEnd` (for the `touchend` event)
+  * `@touchMove` (for the `touchmove` event)
+  * `@touchStart` (for the `touchstart` event)
+  * `@focus-in` (for the `focusin` event)
+  * `@focus-out` (for the `focusout` event)
+  * `@key-down` (for the `keydown` event)
+  * `@key-press` (for the `keypress` event)
+  * `@key-up` (for the `keyup` event)
+
+These arguments are no longer necessary – with angle bracket invocations, DOM
+event listeners can be registered directly using the `{{on}}` modifier. An
+invocation passing one or more of these named arguments now triggers a
+deprecation warning.
+
+Before:
+
+```hbs
+<Input @click={{this.onClick}} />
+       ~~~~~~~~~~~~~~~~~~~~~~~
+or
+
+{{input click=this.onClick}}
+        ~~~~~~~~~~~~~~~~~~
+
+Passing the `@click` argument to <Input> is deprecated. Instead, please use the
+{{on}} modifier, i.e. `<Input {{on "click" ...}} />` instead of
+`<Input @click={{...}} />` or `{{input click=...}}`.
+```
+
+After:
+
+```hbs
+<Input {{on "click" this.onClick}} />
+```
+
+Note that these named arguments were not necessarily an intentional part of the
+component's original design. Rather, these are callbacks that would have fired
+on all classic components, and since classic components' arguments are set on
+the component instances as properties, passing these arguments at invocation
+time would have "clobbered" any callbacks with the same name defined on the
+component's class/prototype, whether it was intended by the component's author
+or not.
+
+For instance, the `<Input>` and `<Textarea>` built-in components implemented
+callbacks that would have been clobbered by these named arguments (may not be a
+complete list):
+
+* `@change`
+* `@focusIn`
+* `@focusOut`
+* `@keyDown`
+* `@keyPress`
+* `@keyUp`
+
+Passing these named arguments historically supressed certain behavior of the
+built-in components, in some cases preventing the components from functioning
+properly. This was never an intended part of the original design and should be
+considered a bug.
+
+This bug may be fixed at any time during the transition period – the supression
```

Comment:
What transition period are we talking about? Ember 3.27 / Ember 3.28? As a reader I’m left with questions about how urgent this is relative to all the other deprecations I have to address

Reply:
Does the prose in the latest commit addresses your concern?

Reply:
@chancancode yes, that's far clearer, thx!